codec-http/src/main/java/io/netty/handler/codec/http/Rfc6265ServerCookieEncoder.java

```diff
@@ -0,0 +1,172 @@
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static io.netty.handler.codec.http.CookieEncoderUtil.*;
+
+/**
+ * A <a href="http://tools.ietf.org/html/rfc6265">RFC6265</a> compliant cookie encoder to be used server side,
+ * so some fields are sent (Version is typically ignored).
+ *
+ * As Netty's Cookie merges Expires and MaxAge into one single field, only Max-Age field is sent.
+ *
+ * Note that multiple cookies are supposed to be sent at once in a single "Set-Cookie" header.
+ *
+ * <pre>
+ * // Example
+ * {@link HttpRequest} req = ...;
+ * res.setHeader("Cookie", {@link Rfc6265ServerCookieEncoder}.encode("JSESSIONID", "1234"));
+ * </pre>
+ *
+ * @see Rfc6265ServerCookieDecoder
+ */
+public final class Rfc6265ServerCookieEncoder {
+
+    /**
+     * Encodes the specified cookie name-value pair into a Set-Cookie header value.
+     *
+     * @param name the cookie name
+     * @param value the cookie value
+     * @return a single Set-Cookie header value
+     */
+    public static String encode(String name, String value) {
+        return encode(new DefaultCookie(name, value));
+    }
+
+    /**
+     * Encodes the specified cookie into a Set-Cookie header value.
+     *
+     * @param cookie the cookie
+     * @return a single Set-Cookie header value
+     */
+    public static String encode(Cookie cookie) {
+        if (cookie == null) {
+            throw new NullPointerException("cookie");
+        }
+
+        StringBuilder buf = stringBuilder();
+
+        addUnquoted(buf, cookie.name(), cookie.value());
+
+        if (cookie.maxAge() != Long.MIN_VALUE) {
+            add(buf, CookieHeaderNames.MAX_AGE, cookie.maxAge());
+        }
+
+        if (cookie.path() != null) {
+            addUnquoted(buf, CookieHeaderNames.PATH, cookie.path());
+        }
+
+        if (cookie.domain() != null) {
+            addUnquoted(buf, CookieHeaderNames.DOMAIN, cookie.domain());
+        }
+        if (cookie.isSecure()) {
+            buf.append(CookieHeaderNames.SECURE);
+            buf.append((char) HttpConstants.SEMICOLON);
+            buf.append((char) HttpConstants.SP);
+        }
+        if (cookie.isHttpOnly()) {
+            buf.append(CookieHeaderNames.HTTPONLY);
+            buf.append((char) HttpConstants.SEMICOLON);
+            buf.append((char) HttpConstants.SP);
+        }
+
+        return stripTrailingSeparator(buf);
+    }
+
+    /**
+     * Batch encodes cookies into Set-Cookie header values.
+     *
+     * @param cookies a bunch of cookies
+     * @return the corresponding bunch of Set-Cookie headers
+     */
+    public static List<String> encode(Cookie... cookies) {
+        if (cookies == null) {
+            throw new NullPointerException("cookies");
+        }
+
+        if (cookies.length == 0) {
+            return Collections.emptyList();
```

Comment:
i actually think returning an empty list is the right thing to do here.  Instead, we'd better fix here: https://github.com/netty/netty/pull/3239/files#r23205278